src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs

```diff
@@ -0,0 +1,151 @@
+                diffEntry.TargetObjectType = targetType.TypeName;
+            }
+
+            if (difference.DifferenceType == SchemaDifferenceType.Object)
+            {
+                // set source and target scripts
+                if (difference.SourceObject != null)
+                {
+                    string sourceScript;
+                    difference.SourceObject.TryGetScript(out sourceScript);
+                    diffEntry.SourceScript = FormatScript(sourceScript);
+                }
+                if (difference.TargetObject != null)
+                {
+                    string targetScript;
+                    difference.TargetObject.TryGetScript(out targetScript);
+                    diffEntry.TargetScript = FormatScript(targetScript);
+                }
+            }
+
+            diffEntry.Children = new List<DiffEntry>();
+
+            foreach (SchemaDifference child in difference.Children)
+            {
+                diffEntry.Children.Add(CreateDiffEntry(child, diffEntry));
+            }
+
+            return diffEntry;
+        }
+
+        internal static SchemaCompareEndpoint CreateSchemaCompareEndpoint(SchemaCompareEndpointInfo endpointInfo, string connectionString)
+        {
+            switch (endpointInfo.EndpointType)
+            {
+                case SchemaCompareEndpointType.Dacpac:
+                    {
+                        return new SchemaCompareDacpacEndpoint(endpointInfo.PackageFilePath);
+                    }
+                case SchemaCompareEndpointType.Database:
+                    {
+                        return new SchemaCompareDatabaseEndpoint(connectionString);
+                    }
+                default:
+                    {
+                        return null;
+                    }
+            }
+        }
+
+        internal static string GetConnectionString(ConnectionInfo connInfo, string databaseName)
+        {
+            if (connInfo == null)
+            {
+                return null;
+            }
+
+            connInfo.ConnectionDetails.DatabaseName = databaseName;
+            return ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
+        }
+
+
+        private static string GetName(string name)
+        {
+            // remove brackets from name
+            return Regex.Replace(name, @"[\[\]]", "");
+        }
+
+
+        internal static string RemoveExcessWhitespace(string script)
+        {
+            if (script != null)
+            {
+                // remove leading and trailing whitespace
+                script = script.Trim();
+                // replace all multiple spaces with single space
+                script = Regex.Replace(script, " {2,}", " ");
+            }
+            return script;
+        }
+
+        internal static string FormatScript(string script)
+        {
+            script = RemoveExcessWhitespace(script);
```

Comment:
Not new code, just moved to utils since it was used by multiple classes. from what I understand it only removes two or more consecutive ones... @kisantia can you please correct my explanation :)

Reply:
Yes it removes multiple consecutive spaces so that the diff in ADS doesn't display it as a difference. Having different whitespace was a problem because spaces are used to align columns, so a table with a new column with a longer name would the whole table as being different instead of just the new column.